CardScan/Classes/CreditCardUtils.swift

```diff
@@ -1,29 +1,43 @@
 import Foundation
 
+public enum CardNetwork: String {
+    case VISA = "Visa"
+    case MASTERCARD = "Mastercard"
+    case AMEX = "Amex"
+    case DISCOVER = "Discover"
+    case UNIONPAY = "Union Pay"
+    case JCB = "Jcb"
+    case DINERSCLUB = "Diners Club"
+    case UNKNOWN = "Unknown"
+}
+
 public struct CreditCardUtils {
     static let maxPanLength = 16
     static let maxPanLengthAmericanExpress = 15
     static let maxPanLengthDinersClub = 14
 
-    static let prefixesAmericanExpress = ["34", "37"]
-    static let prefixesDinersClub = ["300", "301", "302", "303", "304", "305", "309", "36", "38", "39"]
-    static let prefixesDiscover = ["6011", "64", "65"]
-    static let prefixesJcb = ["35"]
-    static let prefixesMastercard = ["2221", "2222", "2223", "2224", "2225", "2226",
+    private static let prefixesAmericanExpress = ["34", "37"]
+    private static let prefixesDinersClub = ["300", "301", "302", "303", "304", "305", "309", "36", "38", "39"]
+    private static let prefixesDiscover = ["6011", "64", "65"]
+    private static let prefixesJcb = ["35"]
+    private static let prefixesMastercard = ["2221", "2222", "2223", "2224", "2225", "2226",
                                      "2227", "2228", "2229", "223", "224", "225", "226",
                                      "227", "228", "229", "23", "24", "25", "26", "270",
                                      "271", "2720", "50", "51", "52", "53", "54", "55",
                                      "67"]
-    static let prefixesUnionPay = ["62"]
-    static let prefixesVisa = ["4"]
+    private static let prefixesUnionPay = ["62"]
+    private static let prefixesVisa = ["4"]
+    
+    
+    public static func isValidNumber(cardNumber: String) -> Bool {
+        return isValidLuhnNumber(cardNumber: cardNumber) && isValidLength(cardNumber: cardNumber)
+    }
     
     // https://en.wikipedia.org/wiki/Luhn_algorithm
     // assume 16 digits are for MC and Visa (start with 4, 5) and 15 is for Amex
     // which starts with 3
-    public static func luhnCheck(_ cardNumber: String) -> Bool {
-        if cardNumber.count == 0 {
-            return false
-        } else if !isValidBin(number: cardNumber) {
+    static func isValidLuhnNumber(cardNumber: String) -> Bool {
```

Comment:
You're taking a function that used to be public and making it not public. If people who use our library use this function they're going to be upset that they need to change code now.